### DISCUSSION TITLE
I2C address for the display

### DISCUSSION THREAD

**Discussion by Pitichinaccio** [srozum/film_camera_tester#4] [General]
Hello Serhiy,

thanks for sharing your great shutter tester. I am just building my own
version and have an issue with the LCD display.
What is the I2C-address used in your code?
I flashed the Arduino and connected a display that uses 0x3C.  It only
flashes very shortly, once about every second (I guess the Arduino just
reboots).

Does your firmware search for different I2C-addresses until it finds the
display? Or do I have to connect the keypad and the sensors first (they
are not ready yet so I just tested the Aduino with the LCD) ?

Bernd

**Comment by srozum** [comment]
Hello,

The screen used in Tester is 2004A,  English/Japaneese version, with default address 0x27. 

I've heard about European/Cyrillic versions, probably that's what you have. These screens have different character sets as well as different special characters. So, it wouldn't work with the tester since it uses some special characters for the user interface, which are not present in European version.

If you will be able to confirm that by testing with another screen, I will update Wiki.

Tester doesn't need any sensor to be connected to start, so most likely yours restarts because of LCD.

Cheers, 
Serhiy

**Reply by Pitichinaccio** [reply]
Hello Serhiy, thanks for your reply.
The LCD I used is an English/Japanese type. But the address is 0x3C.
So I guess it is a problem with the I2C address.
I will have to try to change that then.

Bernd

**Comment by srozum** [comment]
Actually, address of LCD is defined by the chip used on a backpack board, and I have no idea if LCD version (English vs European) is related to what backpack chip is used.

Note that most piggy-back modules are centered around PCF8574T (SO16 package of PCF8574 in DIP16 package) with a default slave address of 0x27. If your piggy-back board holds a PCF8574AT chip, then the default slave address will change to 0x3F. In short, if the piggy-back board is based on PCF8574T and the address connections (A0-A1- A2) are not bridged with solder it will have the slave address 0x27.

**Reply by Pitichinaccio** [reply]
It has an PCF8547T chip but it's default address is 0x3C.

**Comment by Pitichinaccio** [comment]
Seems that I can't change it to 0x27 although it's marked as the T-type ... strange. 
I ordered another one with 0x27 as default, so I hope this will work fine.

Bernd
